Added docs for scripts in FLOSS (#677)
mandiant/flare-floss: scripts/README.md

# FLOSS Scripts
FLOSS supports converting its output into scripts for various tools. Please see the render scripts in this directory.
  
Additionally, there is another [plugin for IDA](idaplugin.py) to allow FLOSS to automatically

> extract obfuscated strings and apply them to the currently loaded module in IDA.

Reviewer: ```suggestion
extract obfuscated strings and apply them to the currently loaded module in IDA. `idaplugin.py` is a IDAPython script you can directly run within IDA Pro (File - Script File... [ALT + F7]).
```

# FLOSS Scripts
FLOSS supports converting its output into scripts for various tools. Please see the render scripts in this directory.
  
Additionally, there is another [plugin for IDA](idaplugin.py) to allow FLOSS to automatically
extract obfuscated strings and apply them to the currently loaded module in IDA.

# Installation
These scripts can be downloaded from the FLOSS [GitHub](https://github.com/mandiant/flare-floss) repository
alongside the source, which is required for the scripts to run.
To install FLOSS as source, see the documentation [here](../doc/installation.md).

# Usage
## Convert FLOSS output for use by other tools

- Run FLOSS on the desired executable with the `-j` or `--json` argument to emit a JSON result
and redirect it to a JSON file.

>     $ floss -j suspicious.exe > floss_results.json

Reviewer: ```suggestion
    `$ floss -j suspicious.exe > floss_results.json`
```

> For Binary Ninja, IDA, Ghidra or Radare2,

Reviewer: ```suggestion
For Binary Ninja, IDA Pro, Ghidra or Radare2:
```

> - Run the script for your tool of choice by passing the result json as an argument and
> redirect the output to a Python(.py) file.  

Reviewer: ```suggestion
- Run the script for your tool of choice by passing the result json file as an argument and
redirect the output to a Python (.py) file.  
```

Ghidra Example:

>     $ python render-ghidra-import-script.py floss_results.json > apply_floss.py

Reviewer: ```suggestion
    `$ python render-ghidra-import-script.py floss_results.json > apply_floss.py`
```

Author: But wouldn't this make it a one line code? I wanted it to look similar to the existing docs.
![image_2023-04-04_132859320](https://user-images.githubusercontent.com/126584775/229726756-e5a52914-23ee-4b1d-90be-e8e4b61ccf2e.png)

> - Run the new python file apply_floss.py in the tool you made it for.

Reviewer: ```suggestion
- Run the Python script `apply_floss.py` using the desired tool.
```

> For x64dbg,

Reviewer: ```suggestion
For x64dbg:
```

> - Instead of a python file, redirect the output to a .json file.

Reviewer: ```suggestion
- Instead of a Python file, redirect the output to a .json file.
```

>     $ python render-x64dbg-database.py floss-results.json > database.json

Reviewer: ```suggestion
    `$ python render-x64dbg-database.py floss-results.json > database.json`
```

> - Open the JSON file database.json in x64dbg.

Reviewer: ```suggestion
- Open the JSON file `database.json` in x64dbg.
```

> # FLARE Obfuscated String Solver
> 
> ## Scripts

Reviewer: ```suggestion
# FLOSS Scripts
```

> FLOSS currently supports converting its output into scripts for the following tools:

Reviewer: ```suggestion
FLOSS supports converting its output into scripts for various tools. Please see the render scripts in this directory.
```

> 1. [IDA pro](render-ida-import-script.py)
> 2. [Binary Ninja](render-binja-import-script.py)
> 3. [Ghidra](render-ghidra-import-script.py)
> 4. [Radare2](render-r2-import-script.py)
> 5. [X64dbg](render-x64dbg-database.py)

Reviewer: ```suggestion
```

Reviewer: I suggest this to reduce overhead keeping this in sync.

  
Additionally, there is another [plugin for IDA](idaplugin.py) to allow FLOSS to automatically
extract obfuscated strings and apply them to the currently loaded module in IDA.

# Installation
These scripts can be downloaded from the FLOSS [GitHub](https://github.com/mandiant/flare-floss) repository
alongside the source, which is required for the scripts to run.
To install FLOSS as source, see the documentation [here](../doc/installation.md).

# Usage
#### To convert FLOSS output into scripts for tools:

- Run FLOSS on the desired executable with the -j or --json argument to emit a JSON result
and redirect it to a JSON file.

> 

Reviewer: ```suggestion
```

    $ floss -j suspicious.exe > floss_results.json

> For Binary Ninja, IDA, Ghidra or Radare2,

Reviewer: I think listing one example each is fine here and reduces potential maintenance.

Author: Thank you, I was wondering a bit about that. I changed it to only have one example. Also sorry for merging all commits separately, I noticed the batch option just a little bit too late.

Reviewer: No worries, we can squash and merge here.
